### PR TITLE
Ad 95/settings

### PR DESCRIPTION
**Pull request by BartoszBilejczyk**

### CODE REVIEW

File: src/app/models/billing-history-item.model.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export interface BillingHistoryItemModel {
+  status: number;
+  date: any;
```
Comment (aqum) [inline src/app/models/billing-history-item.model.ts]: Avoid using `any`. I saw you have string under date in mocked object. I'd expect `Date` object to be there and then in html you can use angular filter to show human redable date.

File: src/app/settings/billing/billing-history/billing-history-withdrawal/billing-history-withdrawal.component.html [hunk 2]
```diff
@@ -0,0 +1,50 @@
+<div class="col-xs-2">
+  <div class="
```
Comment (KonradFak) [inline src/app/settings/billing/billing-history/billing-history-withdrawal/billing-history-withdrawal.component.html]: Move class attribute to new line

File: src/app/settings/billing/billing-history/billing-history.component.html [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        class="
+        adsh-heading
+        adsh-heading--primary"
+      >
+        Notification Settings
+      </h2>
+      <p
+        class="
+        adsh-copy
+        adsh-copy--large"
```
Comment (KonradFak) [inline src/app/settings/billing/billing-history/billing-history.component.html]: Fix indent

File: src/app/settings/billing/billing-history/billing-history.component.html [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+          </app-billing-history-withdrawal>
+        </li>
+      </ul>
+    </div>
+
+    <button
+      class="
+          adsh-btn
+          adsh-btn--white
+          adsh-blue"
```
Comment (KonradFak) [inline src/app/settings/billing/billing-history/billing-history.component.html]: Too much indent

File: src/app/settings/billing/billing-history/billing-history.component.html [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    >
+      <img
+        class="
+            adsh-icon
+            adsh-icon--prepend"
```
Comment (KonradFak) [inline src/app/settings/billing/billing-history/billing-history.component.html]: Too much indent

File: src/app/settings/billing/billing-history/billing-history.component.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+      address: '0x35-0xww9091j-x530',
+      link: 'https://etherscan.io/address/0xde9190f9901',
+    },
+    {
+      status: 349.80,
+      date: '17 Jan 2018',
+      address: '0x35-0xww9091j-x530',
+      link: 'https://etherscan.io/address/0xde9190f9901',
+    },
+  ];
```
Comment (KonradFak) [inline src/app/settings/billing/billing-history/billing-history.component.ts]: Shouldn't this be moved to models dir?

File: src/app/settings/billing/billing.component.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+  selector: 'app-payment',
+  templateUrl: './billing.component.html',
+  styleUrls: ['./billing.component.scss'],
+})
+export class BillingComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit() {
+  }
```
Comment (KonradFak) [inline src/app/settings/billing/billing.component.ts]: Remove unused functions and their imports

File: src/app/settings/billing/user-wallet/user-wallet.component.html [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+                    adsh-icon--prepend"
+                  src="assets/images/credit-card--blue.svg"
+                >
+                  Deposit Funds
+              </button>
+              <button class="adsh-btn adsh-btn--blue">
+                <img
+                  class="
+                      adsh-icon
+                      adsh-icon--prepend"
```
Comment (KonradFak) [inline src/app/settings/billing/user-wallet/user-wallet.component.html]: Too much indent

File: src/app/settings/billing/user-wallet/user-wallet.component.scss [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+
+  .adsh-btn {
+    margin-right: 20px;
+  }
+}
+
+.withdrawal {
+  margin-bottom: 15px;
+
+  >div {
```
Comment (KonradFak) [inline src/app/settings/billing/user-wallet/user-wallet.component.scss]: shouldn't there be whitespace after `>`?

File: src/app/settings/billing/user-wallet/user-wallet.component.ts [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+  templateUrl: './user-wallet.component.html',
+  styleUrls: ['./user-wallet.component.scss'],
+  encapsulation: ViewEncapsulation.None
+})
+export class UserWalletComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit() {
+  }
```
Comment (KonradFak) [inline src/app/settings/billing/user-wallet/user-wallet.component.ts]: Remove unused functions and their imports

File: src/app/settings/billing/user-wallet/user-wallet.component.ts [hunk 11]
```diff
@@ -0,0 +1,16 @@
+import { Component, OnInit, ViewEncapsulation } from '@angular/core';
+
+@Component({
+  selector: 'app-user-wallet',
+  templateUrl: './user-wallet.component.html',
+  styleUrls: ['./user-wallet.component.scss'],
+  encapsulation: ViewEncapsulation.None
```
Comment (KonradFak) [inline src/app/settings/billing/user-wallet/user-wallet.component.ts]: Remove `encapsulation: ViewEncapsulation.None` as we decided to keep styles isolated for each component, leave it default value by not setting this field

File: src/app/settings/general-settings/general-settings.component.ts [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+  styleUrls: ['./general-settings.component.scss'],
+  encapsulation: ViewEncapsulation.None
+})
+export class GeneralSettingsComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit() {
+  }
+
```
Comment (KonradFak) [inline src/app/settings/general-settings/general-settings.component.ts]: Please apply to this component previous comments about unused function and encapsulation

File: src/app/settings/general-settings/notification-settings/notification-settings.component.html [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+        class="
+          adsh-copy
+          adsh-copy--large"
+      >
+        In addition to these alerts, when necessary, we may send you communications about your advertising agreement with Adshares.
+      </p>
+      <ul
+        class="
+          notification-table
+          adsh-copy">
```
Comment (KonradFak) [inline src/app/settings/general-settings/notification-settings/notification-settings.component.html]: Ending `>` should be in new line

File: src/app/settings/general-settings/notification-settings/notification-settings.component.ts [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+      topic: 'Performance Suggestions',
+      component: true,
+      email: false,
+    },
+    {
+      topic: 'Special Offers',
+      component: true,
+      email: true,
+    }
+  ];
```
Comment (KonradFak) [inline src/app/settings/general-settings/notification-settings/notification-settings.component.ts]: Should be mocked in easymock

File: src/app/settings/general-settings/notification-settings/notification-settings.component.ts [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+
+  constructor() { }
+
+  ngOnInit() {
+  }
```
Comment (KonradFak) [inline src/app/settings/general-settings/notification-settings/notification-settings.component.ts]: Remove unused functions and their imports

File: src/app/settings/general-settings/preferences/preferences.component.html [hunk 16]
```diff
@@ -0,0 +1,128 @@
...
+            adsh-copy--large"
+        >
+          We do our best to give you a great experience - we'll be sad to see you leave us.
+        </p>
+        <button
+          class="
+              adsh-btn
+              adsh-btn--white
+              adsh-btn--no-border
+              adsh-blue"
```
Comment (KonradFak) [inline src/app/settings/general-settings/preferences/preferences.component.html]: Too much indent

File: src/app/settings/general-settings/preferences/preferences.component.ts [hunk 17]
```diff
@@ -0,0 +1,16 @@
...
+  templateUrl: './preferences.component.html',
+  styleUrls: ['./preferences.component.scss'],
+  encapsulation: ViewEncapsulation.None
+})
+export class PreferencesComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit() {
+  }
```
Comment (KonradFak) [inline src/app/settings/general-settings/preferences/preferences.component.ts]: Pls apply to this component comments about unused function and encapsulation

File: src/app/settings/settings-navigation/settings-navigation.component.ts [hunk 18]
```diff
@@ -0,0 +1,36 @@
...
+        { name: 'Your wallet'},
+        { name: 'Billing History'}
+      ]
+    }
+  ];
+
+  constructor() { }
+
+  ngOnInit() {
+  }
```
Comment (KonradFak) [inline src/app/settings/settings-navigation/settings-navigation.component.ts]: Remove unused function

File: src/app/settings/settings-routing.module.ts [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+    component: SettingsComponent,
+    children: [
+      {
+        path: 'general',
+        component: GeneralSettingsComponent
+      },
+      {
+        path: 'billing',
+        component: BillingComponent
+      }
```
Comment (KonradFak) [inline src/app/settings/settings-routing.module.ts]: this `children` elements should fit one line? if so pls change it

File: src/app/settings/settings.component.ts [hunk 20]
```diff
@@ -0,0 +1,16 @@
...
+  templateUrl: './settings.component.html',
+  styleUrls: ['./settings.component.scss'],
+  encapsulation: ViewEncapsulation.None
+})
+export class SettingsComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit() {
+  }
```
Comment (KonradFak) [inline src/app/settings/settings.component.ts]: Apply to this component comment about unused functions and encapsulation

File: src/app/settings/settings.module.ts [hunk 21]
```diff
@@ -1,18 +1,45 @@
 import { NgModule } from '@angular/core';
 import { CommonModule } from '@angular/common';
+import { AppCommonModule } from '../common/common.module';
+import { FormsModule } from '@angular/forms';
+
 import { MatDialogModule } from '@angular/material';
+import { SettingsRoutingModule } from './settings-routing.module';
 
-import { WalletDialogComponent } from './dialogs/wallet-dialog/wallet-dialog.component';
+// import { WalletDialogComponent } from './dialogs/wallet-dialog/wallet-dialog.component';
```
Comment (KonradFak) [inline src/app/settings/settings.module.ts]: You left this comment on purpose? also about rest of commented lines in this file

File: src/app/app.component.ts [hunk 22]
```diff
@@ -1,8 +1,14 @@
...
   templateUrl: './app.component.html',
-  styleUrls: ['./app.component.scss']
+  styleUrls: ['./app.component.scss'],
+  animations: [fadeAnimation]
 })
-export class AppComponent { }
+export class AppComponent {
+  public getRouterOutletState(outlet) {
+    return outlet.isActivated ? outlet.activatedRoute : '';
+  }
```
Comment (KonradFak) [inline src/app/app.component.ts]: You can write it in one line: `public getRouterOutletState(outlet) => outlet.isActivated ? outlet.activatedRoute : ''"` also please check if `public` in necessary

File: src/app/settings/billing/billing-history/billing-history.component.html [hunk 23]
```diff
@@ -0,0 +1,72 @@
...
+      </p>
+      <ul
+        class="
+          billing-table
+          adsh-copy"
+      >
+        <li
+          class="
+          row
+          align-center"
```
Comment (KonradFak) [inline src/app/settings/billing/billing-history/billing-history.component.html]: Fix indent

File: src/app/settings/billing/billing-history/billing-history.component.ts [hunk 24]
```diff
@@ -0,0 +1,31 @@
...
+
+import { Subscription } from 'rxjs/Subscription';
+
+@Component({
+  selector: 'app-billing-history',
+  templateUrl: './billing-history.component.html',
+  styleUrls: ['./billing-history.component.scss'],
+})
+export class BillingHistoryComponent implements OnInit {
+  private subscription: Subscription;
```
Comment (KonradFak) [inline src/app/settings/billing/billing-history/billing-history.component.ts]: I think `private` is not necessary here. We should be consistent when writing code and use everywhere in components this operators or not use them

File: src/app/settings/billing/billing.component.ts [hunk 25]
```diff
@@ -0,0 +1,10 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-payment',
+  templateUrl: './billing.component.html',
+  styleUrls: ['./billing.component.scss'],
+})
+export class BillingComponent {
+
+}
```
Comment (KonradFak) [inline src/app/settings/billing/billing.component.ts]: Ig it is empty body for class pls write it as we do it elsewhere: `export class BillingComponent { }`

File: src/app/settings/billing/user-wallet/user-wallet.component.ts [hunk 26]
```diff
@@ -0,0 +1,10 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-user-wallet',
+  templateUrl: './user-wallet.component.html',
+  styleUrls: ['./user-wallet.component.scss']
+})
+export class UserWalletComponent {
+
+}
```
Comment (KonradFak) [inline src/app/settings/billing/user-wallet/user-wallet.component.ts]: Same here with empty body

File: src/app/settings/general-settings/general-settings.component.ts [hunk 27]
```diff
@@ -0,0 +1,10 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-general-settings',
+  templateUrl: './general-settings.component.html',
+  styleUrls: ['./general-settings.component.scss']
+})
+export class GeneralSettingsComponent {
+
+}
```
Comment (KonradFak) [inline src/app/settings/general-settings/general-settings.component.ts]: Also here

File: src/app/settings/general-settings/notification-settings/notification-settings.component.ts [hunk 28]
```diff
@@ -0,0 +1,31 @@
...
+
+import { Subscription } from 'rxjs/Subscription';
+
+@Component({
+  selector: 'app-notification-settings',
+  templateUrl: './notification-settings.component.html',
+  styleUrls: ['./notification-settings.component.scss']
+})
+export class NotificationSettingsComponent implements OnInit{
+  private subscription: Subscription;
```
Comment (KonradFak) [inline src/app/settings/general-settings/notification-settings/notification-settings.component.ts]: Same comment for `private`/`public` operator

File: src/app/settings/general-settings/notification-settings/notification-settings.component.ts [hunk 29]
```diff
@@ -0,0 +1,31 @@
...
+  notificationsSettings: any;
```
Comment (KonradFak) [inline src/app/settings/general-settings/notification-settings/notification-settings.component.ts]: Please use settings interface here

File: src/app/settings/general-settings/preferences/preferences.component.ts [hunk 30]
```diff
@@ -0,0 +1,10 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-preferences',
+  templateUrl: './preferences.component.html',
+  styleUrls: ['./preferences.component.scss']
+})
+export class PreferencesComponent {
+
+}
```
Comment (KonradFak) [inline src/app/settings/general-settings/preferences/preferences.component.ts]: Upper comment for empty body

File: src/app/settings/settings.component.ts [hunk 31]
```diff
@@ -0,0 +1,14 @@
...
+@Component({
+  selector: 'app-settings',
+  templateUrl: './settings.component.html',
+  styleUrls: ['./settings.component.scss'],
+  animations: [fadeAnimation]
+})
+export class SettingsComponent {
+  public getRouterOutletState(outlet) {
+    return outlet.isActivated ? outlet.activatedRoute : '';
+  }
```
Comment (KonradFak) [inline src/app/settings/settings.component.ts]: Pls write this function in one line, and check if public is necessary

File: src/app/store/settings/settings.effects.ts [hunk 32]
```diff
@@ -0,0 +1,35 @@
...
+
+import 'rxjs/add/operator/switchMap';
+
+@Injectable()
+export class SettingsEffects {
+  constructor (
+    private actions$: Actions,
+    private service: SettingsService
+  ) {
+  }
```
Comment (KonradFak) [inline src/app/store/settings/settings.effects.ts]: close body in one line where it starts when empty

File: src/app/store/settings/settings.service.ts [hunk 33]
```diff
@@ -0,0 +1,23 @@
...
+  getBillingHistory(): any {
+    return this.http.get(`${environment.apiUrl}/billing_history`)
+      .map((response: Response) => response.json());
+  }
+
+  getNotificationsSettings(): any {
+    return this.http.get(`${environment.apiUrl}/notifications_settings`)
+      .map((response: Response) => {
+        return response.json();
+      });
```
Comment (KonradFak) [inline src/app/store/settings/settings.service.ts]: Write map function in one line

**Pull request merged**